src/utils.js

```diff
@@ -101,14 +103,49 @@ export function getTabOrigin () {
     return framingOrigin
 }
 
+function findWebsiteOwner (requestData) {
+    // find the site owner
+    const siteUrlList = Array.from(requestData.siteUrlSplit)
+
+    while (siteUrlList.length > 1) {
+        const siteToCheck = siteUrlList.join('.')
+        siteUrlList.shift()
+
+        if (entities[siteToCheck]) {
+            return entities[siteToCheck]
+        }
+    }
+}
+
+/**
+ * Tests whether the two URL's belong to the same
+ * top level domain.
+ */
+function isSameTopLevelDomain (url1, url2) {
```

Comment:
We should probably file a follow up to share as much of this code with the extension.